### PR TITLE
AA: fix timeout when processing multiple incoming requests

### PR DESCRIPTION
**Pull request by imlk0**
The attestation-agent provided two binaries `aa-ttrpc` and `aa-grpc`. However, due to the existing of Mutex, we can't handle multiple `get_token()` and `get_evidence()` requests at the same time.

This commit removes Mutex from `attestation-agent/attestation-agent/src/bin/grpc-aa/server.rs` and `attestation-agent/attestation-agent/src/bin/ttrpc-aa/server.rs` to solve this problem.

### CODE REVIEW

File: attestation-agent/attestation-agent/src/lib.rs [hunk 1]
```diff
@@ -207,7 +214,9 @@ impl AttestationAPIs for AttestationAgent {
             operation,
             content,
         };
-        let alg = self.config.eventlog_config.eventlog_algorithm;
+        let alg = config.eventlog_config.eventlog_algorithm;
+        drop(config);
```
Comment (Xynnn007) [inline attestation-agent/attestation-agent/src/lib.rs:218]: ditto

**Review by Xynnn007 — COMMENTED** [review]
Thanks for this fix and this would greatly promote the concurrent performance! some nits.

File: attestation-agent/attestation-agent/src/lib.rs (hunk 1 above)
Comment (imlk0) [reply attestation-agent/attestation-agent/src/lib.rs]: optimized

File: attestation-agent/attestation-agent/src/lib.rs [hunk 2]
```diff
@@ -106,7 +109,7 @@ impl AttestationAgent {
 
     /// Create a new instance of [AttestationAgent].
     pub fn new(config_path: Option<&str>) -> Result<Self> {
-        let config = match config_path {
+        let config = RwLock::new(match config_path {
```
[the remark this answers is not in the document]
Comment (imlk0) [reply attestation-agent/attestation-agent/src/lib.rs]: updated

**Review by Xynnn007 — APPROVED** [review]

**Pull request merged**